core/src/main/java/com/yahoo/oak/InternalOakMap.java

```diff
@@ -726,7 +648,8 @@ OakUnscopedBuffer getMaxKey() {
     }
 
     // encapsulates finding of the orderedChunk in the skip list and later orderedChunk list traversal
-    private OrderedChunk<K, V> findChunk(K key) {
+    @Override
+    protected BasicChunk<K, V> findChunk(K key, ThreadContext ctx) {
```

Comment:
A child class can refine the object it returns in the methods it overrides. The following compiles and reduces the redundant casting in the other methods.
```suggestion
    @Override
    protected OrderedChunk<K, V> findChunk(K key, ThreadContext ctx) {
```

Reply:
Didn't knew that.